pkg/streamingpromql/aggregations/sum.go

```diff
@@ -89,34 +89,43 @@ func (g *SumAggregationGroup) accumulateHistograms(data types.InstantVectorSerie
 		g.histogramSums = g.histogramSums[:steps]
 	}
 
-	for _, p := range data.Histograms {
-		idx := (p.T - start) / interval
+	for inputIdx, p := range data.Histograms {
+		outputIdx := (p.T - start) / interval
 
-		if g.histogramSums[idx] == invalidCombinationOfHistograms {
+		if g.histogramSums[outputIdx] == invalidCombinationOfHistograms {
 			// We've already seen an invalid combination of histograms at this timestamp. Ignore this point.
 			continue
 		}
 
-		if g.histogramSums[idx] == nil {
+		if g.histogramSums[outputIdx] == nil {
 			if lastUncopiedHistogram == p.H {
 				// We've already used this histogram for a previous point due to lookback.
 				// Make a copy of it so we don't modify the other point.
-				g.histogramSums[idx] = p.H.Copy()
+				g.histogramSums[outputIdx] = p.H.Copy()
 				g.histogramPointCount++
+
+				// Ensure the FloatHistogram instance is not reused when the HPoint slice is reused, as we're retaining a reference to it.
+				data.Histograms[inputIdx].H = nil
+
 				continue
 			}
+
 			// This is the first time we have seen this histogram.
```

Comment:
(nit) I think this clarification makes the below `removeReferencesToRetainedHistogram` clearer why we need to do it. Since we may have seen the histogram before (also applies to avg)
```suggestion
			// We have not previously used this histogram as the start of a point.
```

Reply:
Addressed in f88c488.